tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -287,6 +287,20 @@ func (a *AwsProvider) DestroyWindowsVM(instanceID string) error {
 	return nil
 }
 
+// IsVMRunning checks if the VM is in running state currently and then returns failure
+func (a *AwsProvider) IsVMRunning(instanceID string) error {
+	// Check if the instance exists, if not no need to wait return immediately
+	instance, err := a.GetInstance(instanceID)
+	if err != nil {
+		return err
+	}
+	// 16 represents the running state for the VM in AWS
+	if *instance.State.Code != int64(16) {
```

Comment:
Please use `instance.State.Name` and check against `InstanceStateNameRunning`

Reply:
Done